lib/shopify_cli/context.rb

```diff
@@ -90,7 +90,7 @@ def unknown_os?
 
     # will return true if being launched from a tty
     def tty?
-      !testing? && $stdin.tty?
+      !testing? && !$stdin.closed? && $stdin.tty?
```

Comment:
@molly-yu I ran into an issue trying to call `.tty?` on an already-closed stdin stream. The Ruby interpreter raises a runtime error in that case and the execution aborts.